src/ecdsa/ellipticcurve.py

```diff
@@ -1482,6 +1518,28 @@ def __rmul__(self, other):
         """Multiply point by an integer."""
         return self * other
 
+    def _mul_precompute(self, other):
+        """Multiply point by integer with precomputation table."""
+        X3, Y3, Z3, T3, p, a = 0, 1, 1, 0, self.__curve.p(), self.__curve.a()
+        _add = self._add
+        for X2, Y2, T2 in self.__precompute:
+            if other % 2:
+                if other % 4 >= 2:
```

Comment:
I find `== 3` less confusing here. Could be even flatter:

``` python
            if r == 1:
                other = (other - 1) // 2
                X3, Y3, Z3 = _add(X3, Y3, Z3, X2, Y2, 1, p)
            elif r == 3:
                other = (other + 1) // 2
                X3, Y3, Z3 = _add(X3, Y3, Z3, X2, -Y2, 1, p)
            else:  # 0 or 2
                other //= 2
```

Reply:
actually we probably want
```python
if r == 0 or r == 2:
    other //= 2
elif r == 1:
    ...
```
as r will be zero about 2/3rds of the time and this is inside a tight loop

Reply:
```
0: 2 comparisons -> 1 comparison
1: 1 comparison -> 2 comparisons
2: 2 comparisons -> 2 comparisons
3: 2 comparisons -> 3-4? comparisons
where comparison = interned const int comparison + jump
```

I'd expect the gains to be real minor even with the unevenness you've mentioned, but if you think it's worth it, why not.